pkg/scheduler/internal/cache/interface.go

```diff
@@ -95,21 +96,21 @@ type Cache interface {
 
 	// AddNode adds overall information about node.
 	// It returns a clone of added NodeInfo object.
-	AddNode(node *v1.Node) *framework.NodeInfo
+	AddNode(logger klog.Logger, node *v1.Node) *framework.NodeInfo
 
 	// UpdateNode updates overall information about node.
 	// It returns a clone of updated NodeInfo object.
-	UpdateNode(oldNode, newNode *v1.Node) *framework.NodeInfo
+	UpdateNode(logger klog.Logger, oldNode, newNode *v1.Node) *framework.NodeInfo
```

Comment:
For contextual logging there's no way around adding either klog.Logger or a context to the function.

The guidance is to use klog.Logger when the function doesn't need a context for something else.

Reply:
hm... maybe we should avoid such logging in internal. But handle it outside.

Reply:
Also cc @alculquicondor @Huang-Wei who reviewed the contextual logging before.

Reply:
A logger per instance is indeed problematic. When calling methods of that instance, the context in which the method gets called would be ignored when using that logger.

Reply:
> A logger per instance is indeed problematic. When calling methods of that instance, the context in which the method gets called would be ignored when using that logger.

I don't quite understand this sentence, how should I modify it? @pohly

Reply:
Then I think no need to use contextual logging here.

Reply:
> A logger per instance is indeed problematic. When calling methods of that instance, the context in which the method gets called would be ignored when using that logger.

Does this mean that if we pass the context instead of `schd.logger` we can solve the problem? @pohly

Reply:
Passing either a `Logger` or `Context` into a function is both fine. Which one is better depends on whether the function will ever need a `Context` - if yes, use that, if no, use `Logger`.

A field in a struct is problematic. It leads to confusion when other methods of that struct need to do logging and take a `Context`. Therefore I prefer to avoid them and instead add `Logger` parameters.

For this particular example, `UpdateNode(logger klog.Logger, oldNode, newNode *v1.Node) *framework.NodeInfo` seems fine to me.